src/restruct_poc/solver/calculate_constraint_residual_gradient.hpp

```diff
@@ -194,19 +230,29 @@ struct CalculateConstraintResidualGradient {
             B(i, i) = -1.;
         }
 
-        // B12 = tilde(R1*X0)
-        VecTilde(R1_X0, A);
-        for (int i = 0; i < 3; ++i) {
+        if (cd.type == ConstraintType::kCylindrical) {
+            // B22 = cross(R1 * x0_hat, transpose(R2 * z0_hat))
+            // B22 = cross(R1 * x0_hat, transpose(R2 * y0_hat))
```

Comment:
Don't these comments contradict each other?

Reply:
Yes. B22 is a 2x3 matrix and I was just trying to capture the two rows, but it could be expressed better